internal/typeinfo/arginfo.go

```diff
@@ -43,9 +43,9 @@ func GenerateArgInfo(typeSamples []any) (ArgInfo, error) {
 			}
 			argInfo[t.Name()] = info
 		case reflect.Pointer:
-			return nil, fmt.Errorf("need struct or map, got pointer to %s", t.Elem().Kind())
+			return nil, fmt.Errorf("need non-pointer type, got pointer to %s", t.Elem().Kind())
```

Comment:
Should we make it `supported type` for the error messages to be uniform?

Reply:
In this case its a little bit arbitrary that we don't accept pointers (this code is called by from typebind). I phrased it like that so that its as obvious as possible how to fix the problem. Its not really well documented right now that we don't accept pointers to the types either.

Reply:
Good idea then, keep it as is.